File: sample-app-java/src/main/java/com/cometchat/sampleapp/java/ui/fragments/UsersFragment.java
package com.cometchat.sampleapp.java.ui.fragments;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.cometchat.sampleapp.java.R;
import com.cometchat.sampleapp.java.databinding.FragmentUsersBinding;
import com.cometchat.sampleapp.java.ui.activity.MessagesActivity;
import com.google.gson.Gson;

public class UsersFragment extends Fragment {
    private FragmentUsersBinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentUsersBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        binding.users.setOnItemClick((view1, position, user) -> {
            Intent intent = new Intent(requireActivity(), MessagesActivity.class);
            intent.putExtra(getString(R.string.app_user), new Gson().toJson(user));
            startActivity(intent);
        });
    }
}

File: sample-app-kotlin-push/src/main/java/com/cometchat/sampleapp/kotlin/push/appflow/fragments/UsersFragment.kt
package com.cometchat.sampleapp.kotlin.push.appflow.fragments

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.cometchat.sampleapp.kotlin.push.R
import com.cometchat.sampleapp.kotlin.push.appflow.MessagesActivity
import com.cometchat.sampleapp.kotlin.push.databinding.FragmentUsersBinding
import com.google.gson.Gson

/**
 * Fragment displaying the CometChatUsers component.
 * Mirrors the UsersFragment from master-app-kotlin.
 */
class UsersFragment : Fragment() {

    private var _binding: FragmentUsersBinding? = null
    private val binding get() = _binding!!

    companion object {
        private const val TAG = "UsersFragment"
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentUsersBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        
        // Set up user click listener to navigate to MessagesActivity
        binding.users.setOnItemClick { user ->
            Log.d(TAG, "Navigating to messages with user: ${user.name}")
            val intent = Intent(requireActivity(), MessagesActivity::class.java)
            intent.putExtra(getString(R.string.app_user), Gson().toJson(user))
            startActivity(intent)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
